warehouse/internal/repo/source.go

```diff
@@ -0,0 +1,336 @@
+	); err != nil {
+		return fmt.Errorf("scanning: %w", err)
+	}
+
+	if errorRaw.Valid {
+		sourceJob.Error = errors.New(errorRaw.String)
+	}
+	if jobTypeRaw.Valid {
+		switch jobTypeRaw.String {
+		case model.SourceJobTypeDeleteByJobRunID:
+			sourceJob.JobType = jobTypeRaw.String
+		default:
+			return fmt.Errorf("scanning: unknown job type: %s", jobTypeRaw.String)
+		}
+	} else {
+		return fmt.Errorf("scanning: job type is null")
+	}
+
+	sourceJob.CreatedAt = sourceJob.CreatedAt.UTC()
+	sourceJob.UpdatedAt = sourceJob.UpdatedAt.UTC()
+
+	return nil
+}
+
+func (repo *Source) GetByJobRunTaskRun(
+	ctx context.Context,
+	jobRunID string,
+	taskRunID string,
+) (*model.SourceJob, error) {
+	row := repo.db.QueryRowContext(ctx, `
+		SELECT
+			`+sourceJobColumns+`
+		FROM
+			`+sourceJobTableName+`
+		WHERE
+			metadata->>'job_run_id' = $1 AND
+			metadata->>'task_run_id' = $2
+		LIMIT 1;
+	`,
+		jobRunID,
+		taskRunID,
+	)
+
+	var sourceJob model.SourceJob
+	err := scanSourceJob(row.Scan, &sourceJob)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, model.ErrSourcesJobNotFound
+	}
+	if err != nil {
+		return nil, fmt.Errorf("scanning: %w", err)
+	}
+	return &sourceJob, nil
+}
+
+func (repo *Source) OnUpdateSuccess(
+	ctx context.Context,
+	id int64,
+) error {
+	r, err := repo.db.ExecContext(ctx, `
+		UPDATE
+			`+sourceJobTableName+`
+		SET
+			status = $1,
+			updated_at = $2
+		WHERE
+			id = $3;
+`,
+		model.SourceJobStatusSucceeded,
+		repo.now(),
+		id,
+	)
+	if err != nil {
+		return fmt.Errorf("on update success: %w", err)
+	}
+	rowsAffected, err := r.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("rows affected: %w", err)
+	}
+	if rowsAffected == 0 {
+		return model.ErrSourcesJobNotFound
+	}
+	return nil
+}
+
+func (repo *Source) OnUpdateFailure(
+	ctx context.Context,
+	id int64,
+	error error,
+	maxAttempt int,
+) error {
+	r, err := repo.db.ExecContext(ctx, `
+		UPDATE
+			`+sourceJobTableName+`
+		SET
+ 		  status = (
+			CASE WHEN attempt > $1 THEN $2
+			ELSE $3 END
+		  ),
+		  attempt = attempt + 1,
+		  updated_at = $4,
+		  error = $5
+		WHERE
+			id = $6;
+`,
+		maxAttempt,
+		model.SourceJobStatusAborted,
+		model.SourceJobStatusFailed,
+		repo.now(),
+		error.Error(),
+		id,
+	)
+	if err != nil {
+		return fmt.Errorf("on update failed: %w", err)
+	}
+	rowsAffected, err := r.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("rows affected: %w", err)
+	}
+	if rowsAffected == 0 {
+		return model.ErrSourcesJobNotFound
+	}
+	return nil
+}
+
+func (repo *Source) MarkExecuting(
+	ctx context.Context,
+	ids []int64,
+) error {
+	r, err := repo.db.ExecContext(ctx, `
+		UPDATE
+			`+sourceJobTableName+`
+		SET
+			status = $1,
+			updated_at = $2
+		WHERE
+			id = ANY($3);
+`,
+		model.SourceJobStatusExecuting,
+		repo.now(),
+		pq.Array(ids),
+	)
+	if err != nil {
+		return fmt.Errorf("marking executing: %w", err)
+	}
+	rowsAffected, err := r.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("rows affected: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("mark executing rows affected: %w", err)
```

The same can be done for other errors in this repo. It's not very important, it's just to get a bit more context.
In this specific case the same error with just `"rows affected: %w"` is used in other places as well so it might make it hard to pinpoint where the error is coming from.

Reply:
Adding appropriate information about the action we are performing.